Actions/IncrementVersionNumber/IncrementVersionNumber.ps1

```diff
@@ -47,23 +47,35 @@ try {
     # Collect all projects (AL and PowerPlatform Solution)
     $projectList = @(GetProjectsFromRepository -baseFolder $baseFolder -projectsFromSettings $settings.projects -selectProjects $projects)
     $PPprojects = @(GetMatchingProjects -projects @($settings.powerPlatformSolutionFolder) -selectProjects $projects)
-    if ($projectList.Count -eq 0 -and $PPproject.Count -eq 0) {
-        throw "No projects matches '$selectProjects'"
+    if ($projectList.Count -eq 0 -and $PPprojects.Count -eq 0) {
+        throw "No projects matches '$projects'"
     }
 
+    $repositorySettingsPath = Join-Path $baseFolder $RepoSettingsFile # $RepoSettingsFile is defined in AL-Go-Helper.ps1
+
     # Increment version number in AL Projects
     if ($projectList.Count -gt 0) {
         $allAppFolders = @()
+        $repoVersionExistsInRepoSettings = Test-SettingExists -settingsFilePath $repositorySettingsPath -settingName 'repoVersion'
+        $updateRepoVersionInRepoSettings = $false
         foreach($project in $projectList) {
             $projectPath = Join-Path $baseFolder $project
             $projectSettingsPath = Join-Path $projectPath $ALGoSettingsFile # $ALGoSettingsFile is defined in AL-Go-Helper.ps1
             $settings = ReadSettings -baseFolder $baseFolder -project $project
 
-            # Ensure the repoVersion setting exists in the project settings. Defaults to 1.0 if it doesn't exist.
-            Set-VersionInSettingsFile -settingsFilePath $projectSettingsPath -settingName 'repoVersion' -newValue $settings.repoVersion -Force
-
-            # Set repoVersion in project settings according to the versionNumber parameter
-            Set-VersionInSettingsFile -settingsFilePath $projectSettingsPath -settingName 'repoVersion' -newValue $versionNumber
+            if (Test-SettingExists -settingsFilePath $projectSettingsPath -settingName 'repoVersion') {
+                # If 'repoVersion' exists in the project settings, update it there
+                Set-VersionInSettingsFile -settingsFilePath $projectSettingsPath -settingName 'repoVersion' -newValue $versionNumber
+            } elseif ($repoVersionExistsInRepoSettings) {
+                # If 'repoVersion' is not found in project settings but it exists in repo settings, update it there instead
+                Write-Host "Setting 'repoVersion' not found in $projectSettingsPath. Updating it on repo level instead"
+                $updateRepoVersionInRepoSettings = $true
```

Comment:
At this point the version isn't set, so line 81 (`$projectSettings = ReadSettings -baseFolder $baseFolder -project $project` ) will fetch the old version. That might have effect in `Set-VersionInAppManifests`.

Can you please verify?